### ISSUE TITLE
MPS Crashing On Server Startup with an unsupported and outdated fork

### ISSUE THREAD

**Issue by cybermaxdk** [MachineMuse/MachineMusePowersuits#789]
Hello, when the server start this show up :
cpw.mods.fml.common.LoaderException: java.lang.NoSuchMethodError: extracells.api.definitions.IItemDefinition.wirelessFluidTerminal()Lappeng/api/util/AEItemDefinition;
	at cpw.mods.fml.common.LoadController.transition(LoadController.java:163) ~[LoadController.class:?]
	at cpw.mods.fml.common.Loader.initializeMods(Loader.java:744) ~[Loader.class:?]
	at cpw.mods.fml.server.FMLServerHandler.finishServerLoading(FMLServerHandler.java:97) ~[FMLServerHandler.class:?]
	at cpw.mods.fml.common.FMLCommonHandler.onServerStarted(FMLCommonHandler.java:319) ~[FMLCommonHandler.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:210) ~[lt.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:387) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [li.class:?]
Caused by: java.lang.NoSuchMethodError: extracells.api.definitions.IItemDefinition.wirelessFluidTerminal()Lappeng/api/util/AEItemDefinition;
	at com.qmxtech.powersuitaddons.modules.AppEngWirelessFluidModule.<init>(AppEngWirelessFluidModule.java:42) ~[AppEngWirelessFluidModule.class:?]
	at com.qmxtech.powersuitaddons.common.AddonConfig.loadPowerModules(AddonConfig.java:86) ~[AddonConfig.class:?]
	at com.qmxtech.powersuitaddons.common.ModularPowersuitsAddons.postInit(ModularPowersuitsAddons.java:77) ~[ModularPowersuitsAddons.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_72]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_72]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_72]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_72]
	at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532) ~[FMLModContainer.class:?]
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_72]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_72]
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:275) ~[guava-17.0.jar:?]
	at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212) ~[LoadController.class:?]
	at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190) ~[LoadController.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_72]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_72]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_72]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_72]
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:275) ~[guava-17.0.jar:?]
	at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119) ~[LoadController.class:?]
	at cpw.mods.fml.common.Loader.initializeMods(Loader.java:742) ~[Loader.class:?]
	... 5 more

thx in advance

**Comment by lehjr** [comment]
QMX version is not supported. Will not be supported here.

**Comment by cybermaxdk** [comment]
So which version i'm suppose to download ?

**Comment by lehjr** [comment]
First, never submit bug reports for outdated versions or forks done by someone else. Instead check to see if there is an updated version. Why? Because chances are the bug you are reporting has already been fixed. Look on Curseforge for the latest version.

**Comment by eyeonus** [comment]
"If your comment has been deleted, your answer lies in the readme.", is what I use on my L4D2 map's page.

I don't know what the equivalent would be here.

**Comment by lehjr** [comment]
Second, never submit partial bug reports with no version info. Developers hate that because they have no way of knowing what version you are reporting. The only thing in your bug report that even caught my eye was "QMX", which is a fork of this mod that has not been touched in over 2 years and is based on code even older. The odds of being able to drop that into a Minecraft installation with a bunch of newly updated mods is insanely small. There's a reason I spent so many hours updating this mod and keeping pace with others. Things change and you have to change with them or things will break. It's that simple.

**Comment by lehjr** [comment]
@aionys seems the equivalent here would depend on the mood of the person answering, but generally it would be something equivalent to "don't waste my time or yours reporting outdated versions of the mod. Instead, try the latest version to see if the issue has been fixed. If the issue persists, then report the issue with as much detail as possible."  which I believe is more or less what the readme states which is why I don't put much time into the readme anymore when no one reads it, just like the changelog for the download page.

**Comment by lehjr** [comment]
Oh, and those addons were merged into the mod a year and a half ago, so there is no compatible version of the addons for any version after that.

**Comment by eyeonus** [comment]
How about an open issue marked [super important] [won't fix], titled "If your issue has been closed without reply, review the readme on the [Main Page](https://github.com/MachineMuse/MachineMusePowersuits) before re-submitting or re-opening."

**Comment by eyeonus** [comment]
The whole purpose of the readme is to not have to answer the same question over and over again. It does require a bit extra for some extreme scenarios. Look at me still managing to be polite in reference to them. It's still less effort than dealing with the huge myriad of, bluntly, morons, on an individual basis.

**Comment by lehjr** [comment]
But if they read the readme in the first place, issues like this usually don't appear to begin with. Generally I don't mind responding to these kinds of issues, but my responses are usually better if the person has at least put a little effort into the bug report instead of just copy and pasting a small section of the log and saying something along the lines of "It doesn't work. Plz HELP!!"

With this particular issue, the only reason I know what is causing the crash is because I saw the "QMX" in the crash report and I know how outdated that fork is, and i know that the addons haven't been needed in well over a year because I merged them in, mainly because no one was committed to maintaining them. But beyond that, I have no idea of the version of anything the user has installed.

**Comment by eyeonus** [comment]
Right. That's what I'm talking about. For those people, I have "If your comment has been deleted, your answer lies in the readme."

Someone does a "No work PLZ HLP!" on me, I delete the comment. Either they see that message, read the readme like they should've done in the first place, and fix it their own selves, or they don't comment again. Either way, issue dealt with quick and easy.

That's why I suggested the super important won't fix open issue above, as an equivalent Staples easy button for you on this thing.
Someone opens a "PLZ HLP!" issue, you close without reply, either they take the hint and read the readme or they go continue being oblivious somewhere else.

**Comment by lehjr** [comment]
Issue really needs a "broken user" label, or something else related to "Problem Exists Between Chair And Keyboard"

**Comment by aionys** [comment]
This isn't a won't fix. This is a not a bug.

**Comment by eyeonus** [comment]
They'd have to figure out a way to say it politely in about 50+ different languages....

**Comment by lehjr** [comment]
LOL. Either way, this is simply an issue that the user can solve by updating and not trying to install the addons that were merged in the summer of 2015. This is one of the reasons I stopped supporting 1.7.10. Too many old and outdated mods and too many updated ones. People think that 1.7.10 is just one version of forge and that every release of a mod plays nice with every other release of another mod, regardless of where in the lifespan of 1.710 these mods were updated. 1.7.10 was a long life version, but it went through many breaking changes in both Forge and in mods themselves.
